3.0/m_roleplay.cpp

```diff
@@ -0,0 +1,704 @@
+Send a message to the channel as if it came from the =Narrator= user,
+but send it as a CTCP action (as if =Narrator= used /me).
+
+Note that your nick will be sent with the hostmask of the =Narrator=
+user.
+
+Depending on the server configuration, you may need +o or above in the
+channel, or +U may need to be set.
+">
+
+<helpop key="scene" title="/SCENE <channel> :<message>" value="
+Send a message to the channel as if it came from the =Scene= user.
+
+Note that your nick will be sent with the hostmask of the =Scene=
+user.
+
+Depending on the server configuration, you may need +o or above in the
+channel, or +U may need to be set.
+">
+
+<helpop key="scenea" title="/SCENEA <channel> :<message>" value="
+Send a message to the channel as if it came from the =Scene= user,
+but send it as a CTCP action (as if =Narrator= used /me).
+
+Note that your nick will be sent with the hostmask of the =Scene=
+user.
+
+Depending on the server configuration, you may need +o or above in the
+channel, or +U may need to be set.
+">
+
+<helpop key="npc" title="/NPC <channel> <user> :<message>" value="
+Send a message to the channel as if it came from the given user.
+
+The user given will be sent formatted as underlined to avoid apparent
+spoofing.
+
+Depending on the server configuration, you may need +o or above in the
+channel, or +U may need to be set.
+">
+
+<helpop key="npca" title="/NPCA <channel> <user> :<message>" value="
+Send a message to the channel as if it came from the given user, but
+send it as a CTCP action (as if the given user used /me).
+
+The user given will be sent formatted as underlined to avoid apparent
+spoofing.
+
+Depending on the server configuration, you may need +o or above in the
+channel, or +U may need to be set.
+">
+
+ * Helpop lines for the COPER section
+ * Find: '<helpop key="coper" title="Oper Commands" ...'
+ * Place 'FACTION' and 'FSAY' in the command list accordingly. Re-space as
+ * needed.
+ * Find: '<helpop key="filter" ...'
+ * Place just above that line:
+<helpop key="faction" title="/FACTION <channel> <user> :<message>" value="
+Send a message to the channel as if it came from the given user, but
+send it as a CTCP action (as if the user used /me).
+
+Unlike /NPC, the user will not be underlined.
+
+This command requires the channels/roleplay permission.
+">
+
+<helpop key="fsay" title="/FSAY <channel> <user> :<message>" value="
+Send a message to the channel as if it came from the given user.
+
+Unlike /NPC, the user will not be underlined.
+
+This command requires the channels/roleplay permission.
+">
+
+ * Helpop lines for the CHMODES section
+ * Find: '<helpop key="chmodes" title="Channel Modes" ...'
+ * Change the mode letter if needed and place the following in the channel mode
+ * list accordingly.
+
+ U                  Enable roleplay commands on this channel (requires
+                    the roleplay module)
+ */
+
+#include "inspircd.h"
+
+enum RoleplayNumerics : uint16_t
```

Comment:
This is a C++11 extension.

```suggestion
enum RoleplayNumerics
```